adapter.go

```diff
@@ -94,9 +88,6 @@ func NewAdapter(driverName string, dataSourceName string, dbSpecified ...bool) (
 		return nil, err
 	}
 
-	// Call the destructor when the object is released.
-	runtime.SetFinalizer(a, finalizer)
```

Comment:
Why remove it?

Reply:
I think I did it at the time due to gorm changing the default closure statement for the database(I assumed gorm/v2 did not need a Close() call). I believe I misinterpreted it, and after reading https://github.com/go-gorm/gorm/issues/3145, I have returned a modified finalizer.